src/Microsoft.AspNet.Mvc.Testing/ControllerTestHelperContext.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core.Collections;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+
+namespace Microsoft.AspNet.Mvc.Testing
+{
+	public class ControllerTestHelperContext
+	{
+
+		// Not sure if this RouteData property will be usefull..
+		public RouteData RouteData
+		{
+			get;
+			set;
+		}
+
+		// Not sure if this HttpContext property will be usefull..
+		public HttpContext HttpContext
+		{
+			get;
+			set;
+		}
+
+		public IServiceCollection ServiceCollection
+		{
+			get;
+			set;
+		}
+
+		public IServiceProvider ServiceProvider
+		{
+			get;
+			set;
+		}
```

Comment:
@rynowak I think this design has a problem when users call different overloaded Action methods in a controller. 1) It would make us to maintain a list of delegates for overloaded methods, 2) it makes user to write different callbacks for different overloaded delegates. 
Instead, I'd like to create a context class like
public class UrlHelperContext
{
    string Action {get;set;} string Controller {get; set;...}
}
And ..public Func<UrlHelperContext, string> OnUrlAction { get; set; }

IUrlHeper.RouteUrl method requires the same parameters, I think it makes APIs simpler after all. 

Let me know if you have any concerns on this.